test/e2e_node/mirror_pod_grace_period_test.go

```diff
@@ -131,6 +131,56 @@ var _ = SIGDescribe("MirrorPodWithGracePeriod", func() {
 			framework.ExpectEqual(pod.Spec.Containers[0].Image, image)
 		})
 
+		ginkgo.Context("and the container runtime is temporarily down during pod termination [NodeConformance] [Serial] [Disruptive]", func() {
+			ginkgo.It("the mirror pod should terminate successfully", func(ctx context.Context) {
+				ginkgo.By("delete the static pod")
+				err := deleteStaticPod(podPath, staticPodName, ns)
+				framework.ExpectNoError(err)
+
+				// Note it is important we have a small delay here as we would like to reproduce https://issues.k8s.io/113091 which requires a failure in syncTerminatingPod()
+				// This requires waiting a small period between the static pod being deleted so that syncTerminatingPod() will attempt to run
+				ginkgo.By("sleeping before stopping the container runtime")
+				time.Sleep(2 * time.Second)
+
+				ginkgo.By("stop the container runtime")
+				err = stopContainerRuntime()
+				framework.ExpectNoError(err, "expected no error stopping the container runtime")
+
+				ginkgo.By("waiting for the container runtime to be stopped")
+				gomega.Eventually(ctx, func(ctx context.Context) error {
+					_, _, err := getCRIClient()
+					return err
+				}, 2*time.Minute, time.Second*5).ShouldNot(gomega.Succeed())
+
+				ginkgo.By("start the container runtime")
+				err = startContainerRuntime()
+				framework.ExpectNoError(err, "expected no error starting the container runtime")
+				gomega.Consistently(ctx, func(ctx context.Context) error {
```

Comment:
Wait, if the pod is terminated, shouldn't the mirror pod go away relatively fast?

Shouldn't this treat "no mirror pod" as the success condition?

Reply:
Should. we also be verifying that while the container runtime is down, the mirror pod is listed as still existing?

Reply:
You're right -- I think the issue here is that this check is in the wrong place -- this needs to be before the `startContainerRuntime`. i.e. while the container runtime is stopped the mirror pod should be running, after we start the container runtime, it should be removed.

Reply:
I made a slight tweak - I wanted to verify it at multiple places.  See my change in the metrics commit in my PR.  I think at this point once the tests pass we can keep those commits up to date in my PR vs having the proven failing one here.